task_02/nikolai_chernyshov/main.cpp

```diff
@@ -0,0 +1,175 @@
+            }
+            else (*this->edges_from_ptr[0].ptr).to_json_vert(file);
+        }
+    }
+
+    void to_json_edge(std::ofstream& file) {
+        if (this->edges_from_num > 0) {
+            if (this->edges_from_num > 1) {
+                for (int i=0; i < this->edges_from_num - 1; i++) {
+                    file << "{\n\t\t\t\"id\": " << this->edges_from_ptr[i].edge_id;
+                    file << ",\n\t\t\t\"vertex_ids\": [" << this->id << ", ";
+                    file << this->edges_from_ptr[i].to_vertice << "]\n\t\t}, ";
+                }
+            }
+            file << "{\n\t\t\t\"id\": " << this->edges_from_ptr[this->edges_from_num - 1].edge_id;
+            file << ",\n\t\t\t\"vertex_ids\": [" << this->id << ", ";
+            file << this->edges_from_ptr[this->edges_from_num - 1].to_vertice << "]\n\t\t}";
+
+            if (this->edges_from_num > 0) {
+                bool has_grandkids = false;
+                for (int i=0; i < this->edges_from_num; i++) {
+                    if ((*this->edges_from_ptr[i].ptr).edges_from_num > 0) {
+                        has_grandkids = true;
+                    }
+                }
+                if (has_grandkids) {
+                    file << ", ";
+                }
+            }
+
+            if (this->edges_from_num > 1) {
+                for (int i=0; i < this->edges_from_num - 1; i++) {
+    
+                    if ((*this->edges_from_ptr[i].ptr).edges_from_num > 0) {
+                        (*this->edges_from_ptr[i].ptr).to_json_edge(file);
+                        bool other_has_grandkids = false;
+                        for (int j=i+1; j < this->edges_from_num; j++) {
+                            if ((*this->edges_from_ptr[j].ptr).edges_from_num > 0) {
+                                other_has_grandkids = true;
+                            }
+                        }
+                        if (other_has_grandkids) {
+                            file << ", ";
+                        }
+                    }
+                }
+                if ((*this->edges_from_ptr[this->edges_from_num - 1].ptr).edges_from_num > 0) {
+                    (*this->edges_from_ptr[this->edges_from_num - 1].ptr).to_json_edge(file);
+                }
+            }
+            else if ((*this->edges_from_ptr[0].ptr).edges_from_num > 0) {
+                (*this->edges_from_ptr[0].ptr).to_json_edge(file);
+            }
+        }
+    }
+
+    void to_json(std::ofstream& file) {        
+        file << "{\n\t \"vertices\": [\n\t\t";
+        this->to_json_vert(file);
+        file << "\n\t],\n\t\"edges\": [\n\t\t";
+        this->to_json_edge(file);
+        file << "\n\t]\n}";
+    }
+};
+
+int vertice::vertice_id_max = 0;
+
+int main() {
+    std::shared_ptr<struct vertice> graph_root_ptr(new struct vertice(1, -1));
+
+    std::ofstream myfile;
+    myfile.open("graph.json");
+    (*graph_root_ptr).to_json(myfile);
```

Comment:
В таких случаях используется такая запись:
```suggestion
    graph_root_ptr->to_json(myfile);
```